### PR TITLE
ref(monitors): Prefer slugs over guids

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Updates tests, help strings, and some small bits of logic to prefer calling monitor identifiers 'slugs' instead of simply IDs. This is in line with what we're doing for #1537

I would like to wait for the follow up (support DSN auth) to be ready before merging this, as you will ONLY be able to use legacy GUID/ID identifiers for monitors when using toke auth

### CODE REVIEW

File: src/commands/monitors/run.rs [hunk 1]
```diff
@@ -13,10 +12,10 @@ pub fn make_command(command: Command) -> Command {
     command
         .about("Wraps a command")
         .arg(
-            Arg::new("monitor")
-                .help("The monitor ID")
-                .required(true)
-                .value_parser(Uuid::parse_str),
+            Arg::new("monitor_slug")
+                .value_name("monitor-slug")
```
Comment (kamilogorek) [inline src/commands/monitors/run.rs:16]: This is a breaking change and we need to preserve a hidden alias that'd still support UUID.

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: src/commands/monitors/run.rs (hunk 1 above)
Comment (evanpurkhiser) [reply src/commands/monitors/run.rs:16]: all slugs on the backend match existing UUIDs, it is fully backwards compatible.

Comment (evanpurkhiser) [reply src/commands/monitors/run.rs:16]: See https://github.com/getsentry/sentry/issues/45166

Comment (kamilogorek) [reply src/commands/monitors/run.rs:16]: Ah, perfect 😌

**Pull request merged**